internal/services/dataprotection/data_protection_backup_policy_postgresql_resource.go

```diff
@@ -365,20 +368,29 @@ func expandBackupPolicyPostgreSQLTaggingCriteriaArray(input []interface{}) *[]da
 	}
 	for _, item := range input {
 		v := item.(map[string]interface{})
-		results = append(results, dataprotection.TaggingCriteria{
-			Criteria:        expandBackupPolicyPostgreSQLCriteriaArray(v["criteria"].([]interface{})),
+		result := dataprotection.TaggingCriteria{
 			IsDefault:       utils.Bool(false),
 			TaggingPriority: utils.Int64(int64(v["priority"].(int))),
 			TagInfo: &dataprotection.RetentionTag{
 				ID:      utils.String(v["name"].(string) + "_"),
 				TagName: utils.String(v["name"].(string)),
 			},
-		})
+		}
+		criteria, err := expandBackupPolicyPostgreSQLCriteriaArray(v["criteria"].([]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		result.Criteria = criteria
+
+		results = append(results, result)
 	}
-	return &results
+	return &results, nil
 }
 
-func expandBackupPolicyPostgreSQLCriteriaArray(input []interface{}) *[]dataprotection.BasicBackupCriteria {
+func expandBackupPolicyPostgreSQLCriteriaArray(input []interface{}) (*[]dataprotection.BasicBackupCriteria, error) {
+	if len(input) == 0 || input[0] == nil {
+		return nil, fmt.Errorf("criteria is a required field, cannot leave blank")
+	}
```

Comment:
rather than doing this, we can add `at least one of` to the schema for this block - can we update this?

Reply:
Hi @tombuildsstuff , looks like `AtLeastOneOf` can only be used with `TypeList` has `MaxItems: 1`. But here the parent block `retention_rule` can have multiple items, we cannot reference the attribute under it with `AtLeastOneOf`.

Reply:
Hi @tombuildsstuff , looks like `AtLeastOneOf` can only be used with `TypeList` has `MaxItems: 1`. But here the parent block `retention_rule` can have multiple items, we cannot reference the attribute under it with `AtLeastOneOf`.